File: backend/src/main/java/com/elbuensabor/entities/Pedido.java
package com.elbuensabor.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "pedido")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Pedido extends Base {
    @Column
    private Long fecha;
    @Column
    private Integer numero;
    @Column
    private Long horaEstimadaFin;
    @Column
    private Double total;

    @ManyToOne
    @JoinColumn(name = "envio_id")
    private Envio envio;

    @OneToOne(mappedBy = "pedido")
    private MercadoPagoDatos mercadoPagoDatos;

    @OneToOne(mappedBy = "pedido")
    private Factura factura;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @OneToMany(mappedBy = "pedido")
    private List<DetallePedido> detallesPedido = new ArrayList<DetallePedido>();

    @ManyToOne
    @JoinColumn(name = "domicilio_id")
    private Domicilio domicilio;

    @ManyToOne
    @JoinColumn(name = "estado_id")
    private Estado estado;
}

File: frontend/el-buen-sabor-frontend/src/app/entidades/Pedido.ts
import { Cliente } from './Cliente';
import { DetallePedido } from './DetallePedido';
import { Domicilio } from './Domicilio';
import { Envio } from './Envio';
import { Estado } from './Estado';
import { Factura } from './Factura';
import { MercadoPagoDatos } from './MercadoPagoDatos';

export class Pedido {
  id: number = 0;
  fecha: Date = new Date();
  numero: number = 0;
  horaEstadimadaFin: number = 0;
  total: number = 0;
  envio: Envio = new Envio();
  mercadoPagoDatos: MercadoPagoDatos = new MercadoPagoDatos();
  factura: Factura = new Factura();
  cliente!: Cliente;
  detallesPedido: DetallePedido[] = [];
  domicilio: Domicilio = new Domicilio();
  estado: Estado = new Estado();
}
